src/types.h

```diff
@@ -97,6 +98,9 @@ class SizedType
 
 private:
   bool is_signed_ = false;
+  SizedType *element_type_ = nullptr; // for "container" and pointer
```

Comment:
Isn't this a memory leak?

Hmm, I'd still rather store by value. It looks like `SizedType` could just have some `= default`'d copy constructors and the compiler should be able to do the right thing.

Reply:
> Hmm, I'd still rather store by value. It looks like SizedType could just have some = default'd copy constructors and the compiler should be able to do the right thing.

As in `SizedType elment_type_` or something else? 

> Isn't this a memory leak?

Yes it is. I could go (back) to shared_ptr as we allow copying this object. Or we could go to a unique_ptr (or normal pointer) and copy the object in the copy constructor and make sure it is deleted properly.

Reply:
> As in SizedType elment_type_ or something else?

Yeah, like that.

> Yes it is. I could go (back) to shared_ptr as we allow copying this object. Or we could go to a unique_ptr (or normal pointer) and copy the object in the copy constructor and make sure it is deleted properly.

I'd prefer the method mentioned above. But I also don't wanna block this PR b/c how we store `element_type_` isn't really that important. Whatever seems reasonable to you is fine w/ me.

Reply:
> Yeah, like that.

Hmm I don't think that will be possible as an object storing itself would make it endless nested.

> Whatever seems reasonable to you is fine w/ me.

I think this will work for now. Not entirely sure what the end result will be yet but it should be easy to refactor if needed :)

Reply:
> Hmm I don't think that will be possible as an object storing itself would make it endless nested.

Ah right. Whoops